qiskit_ibm_runtime/transpiler/passes/basis/fold_rzz_angle.py

```diff
@@ -244,3 +248,163 @@ def _quad4(angle: float, qubits: Tuple[Qubit, ...]) -> DAGCircuit:
             check=False,
         )
         return new_dag
+
+
+def convert_to_rzz_valid_pub(
+    program_id: str, pub: Union[SamplerPubLike, EstimatorPubLike]
+) -> Union[SamplerPub, EstimatorPub]:
+    """
+    Return a pub which is compatible with Rzz constraints.
+
+    Current limitations:
+    1. Does not support dynamic circuits.
+    2. Does not preserve global phase.
+    """
+    if program_id == "sampler":
+        pub = SamplerPub.coerce(pub)
+    elif program_id == "estimator":
+        pub = EstimatorPub.coerce(pub)
+    else:
+        raise ValueError(f"Unknown program id {program_id}")
+
+    val_data = pub.parameter_values.data
+    pub_params = np.array(list(chain.from_iterable(val_data)))
+    # first axis will be over flattened shape, second axis over circuit parameters
+    arr = pub.parameter_values.ravel().as_array()
+
+    new_circ = pub.circuit.copy_empty_like()
+    new_data = []
+    rzz_count = 0
+
+    for instruction in pub.circuit.data:
+        operation = instruction.operation
+
+        if operation.name in CONTROL_FLOW_OP_NAMES:
+            raise ValueError(
+                "The function convert_to_rzz_valid_pub currently does not support dynamic instructions."
+            )
+
+        if operation.name != "rzz" or not isinstance(
+            (param_exp := instruction.operation.params[0]), ParameterExpression
+        ):
+            new_data.append(instruction)
+            continue
+
+        param_names = [param.name for param in param_exp.parameters]
+
+        col_indices = [np.where(pub_params == param_name)[0][0] for param_name in param_names]
+        # col_indices is the indices of columns in the parameter value array that have to be checked
```

Comment:
super minor suggestion:

```suggestion
        # col_indices is the indices of columns in the parameter value array that have to be checked
        col_indices = [np.where(pub_params == param_name)[0][0] for param_name in param_names]
```